### ISSUE TITLE
[Source Maps] **seems** to generate misaligned source map

### ISSUE THREAD

**Issue by steveetm** [webpack-contrib/uglifyjs-webpack-plugin#175]
Hi!

I've created a sample repository to demonstrate the problem which you can check [here](https://github.com/steveetm/webpackuglifybugrepro).

Open the generated page in chrome, open devtools and put a breakpoint at console.log('a');
When the debugger stops there, try to step one line, all console logs run instantly and the debugger has no clue on which line it should be.

But if you comment out `new UglifyJsPlugin({sourceMap: true})` from `webpack.config.js` and
run 
`./node_modules/webpack/bin/webpack.js`
`./node_modules/uglifyjs-webpack-plugin/node_modules/uglify-es/bin/uglifyjs ./bundle.js --source-map 'content="bundle.js.map",includeSources,url="bundle.js.map"' -o ./bundle.js -m`
all is fine, you can place your breakpoint in ./webpack/main.js, and go through the source line-by-line.

**Comment by kzc** [comment]
This ought to be in a FAQ for this plugin...

Uglify optimizes code in order to reduce it. Unused code is eliminated and some of the original variables and functions are inlined and as such have no mapping. Breakpoints cannot necessarily be set on certain lines as they no longer exist.
```js
$ cat main.js 
debugger;
console.log('a');
console.log('b');
/**
*
*
*
*/
function test(...args) {
console.log('e');
debugger;
const b = args;
console.log('f');
}
console.log('c');
console.log('d');

test();
```
```
$ bin/uglifyjs -V
uglify-es 3.2.0
```
```js
$ cat main.js | bin/uglifyjs --toplevel -m -c
console.log("a"),console.log("b"),console.log("c"),console.log("d"),console.log("e"),console.log("f");
```

The Chrome debugger cannot handle source maps as well as FireFox. The FireFox debugger can often step through uglified code when Chrome cannot. Report such issues to the Chrome project.

If you want minified code with good debugging in Chrome you can disable uglify `compress` and only `mangle` the code. This will make the uglified output larger.

**Comment by steveetm** [comment]
You were absolutely right, I also missed the -c from my raw uglify command. But our original issue with our whole codebase was if I set a breakpoint on a certainly existing line, the debugger stopped somewhere else instead. While in this example this problem indeed non-existing. I just wrongly assumed this is the exact same issue, but I've failed to reproduce it, as I could easily put a breakpoint e.g. at console.log('d') and the debugger stops there(hitting step-by-step here can jump over lines, I can understand that). Our symptom is that I put a breakpoint a console.log('d') and the debugger stops at console.log('f'), but I can not reproduce it yet.
